### DISCUSSION TITLE
Community Strategies

### DISCUSSION THREAD

**Discussion by sergiodxa** [sergiodxa/remix-auth#111] [Show and tell]
This discussion is to keep a list of any strategy built by the community. Reply below with the strategies you built!

- [OAuth2Strategy](https://github.com/sergiodxa/remix-auth-oauth2)
- [FormStrategy](https://github.com/sergiodxa/remix-auth-form)
- [GitHubStrategy](https://github.com/sergiodxa/remix-auth-github)

**Comment by pbteja1998** [comment]
- [GoogleStrategy](https://github.com/pbteja1998/remix-auth-google) 
- [EmailLinkStrategy]( https://github.com/pbteja1998/remix-auth-email-link)

**Reply by pskd73** [reply]
Would be great if EmailLinkStrategy supports remix-auth@4.1.0 :)

**Reply by nichtsam** [reply]
> Would be great if EmailLinkStrategy supports remix-auth@4.1.0 :)

I’ve forked and updated it with some changes to the usage. 
Feel free to try it out:
https://github.com/nichtsam/remix-auth-email-link

**Reply by btomaj** [reply]
I built a `MagicLinkStrategy` to replace `EmailLinkStrategy` to be more closely aligned with the design intent of `remix-auth@4.1.0` and be usable for SMS, email, etc.: https://github.com/btomaj/remix-auth-magic-link

Still trying to get it published on npm.

**Reply by nichtsam** [reply]
> I built a `MagicLinkStrategy` to replace `EmailLinkStrategy` to be more closely aligned with the design intent of `remix-auth@4.1.0` and be usable for SMS, email, etc.: https://github.com/btomaj/remix-auth-magic-link
> 
> Still trying to get it published on npm.

Niceee, I couldn't access your repo, is it public?

**Reply by btomaj** [reply]
@nichtsam ah, thank you for that! I forgot to make it public. Public now!

**Comment by TheRealFlyingCoder** [comment]
[Remix Auth Socials](https://www.npmjs.com/package/remix-auth-socials) - A Collection of Oauth2 Social login strategies.

Currently included:

Facebook
Google
Discord
Github
Microsoft

**Comment by danestves** [comment]
- [Auth0Strategy](https://github.com/danestves/remix-auth-auth0)

**Comment by juhanakristian** [comment]
[MicrosoftStrategy](https://github.com/juhanakristian/remix-auth-microsoft)

**Comment by mitchelvanbever** [comment]
[supabaseStrategy](https://github.com/mitchelvanbever/remix-auth-supabase-strategy)

**Reply by sindras** [reply]
For future readers, note that this supabase strategy is [no longer supported](https://github.com/mitchelvanbever/remix-auth-supabase/issues/45#issuecomment-1912783915).

**Comment by Andreychik32** [comment]
[SteamStrategy](https://github.com/Andreychik32/remix-auth-steam)

**Reply by Shigbeard** [reply]
Is this still being maintained? The dependencies of this project conflicts with remix now, and there's a pending pull request and issue that havent been acknowledged in months.

**Reply by pedro-santos21** [reply]
Not sure if it's maintained but there are public forks that were updated recently and work for me, such as this one: https://github.com/ianlucas/remix-auth-steam

**Comment by doncicuto** [comment]
- [GitlabStrategy](https://github.com/doncicuto/remix-auth-gitlab). [NPM package](https://www.npmjs.com/package/@doncicuto/remix-auth-gitlab).
- [BitbucketStrategy](https://github.com/doncicuto/remix-auth-bitbucket). [NPM Package](https://www.npmjs.com/package/@doncicuto/remix-auth-bitbucket)

**Comment by josteinkringlen** [comment]
[SpotifyStrategy](https://github.com/JosteinKringlen/remix-auth-spotify)

**Comment by na2hiro** [comment]
[TwitterStrategy](https://github.com/na2hiro/remix-auth-twitter) (OAuth 2.0 and 1.0a)

**Reply by danestves** [reply]
Have you tested that strategy on Cloudflare Workers? I don't think `crypto-js` will work there

**Reply by pbteja1998** [reply]
> Have you tested that strategy on Cloudflare Workers? I don't think `crypto-js` will work there

`crypto-js` does work on CF workers, even I use it in `Email Link Strategy` and it's working.

**Reply by danestves** [reply]
Perfect! I just want to make sure that have support for it

**Reply by na2hiro** [reply]
I chose `crypto-js` because it doesn't seem to `require` anything built-in to Node.js from source code, other than `crypto` which is used only when possible. https://github.com/brix/crypto-js/search?q=require I haven't tested it yet though.

Made an issue there to double check that! https://github.com/na2hiro/remix-auth-twitter/issues/1

**Reply by na2hiro** [reply]
I confirmed it works on Cloudflare Workers: https://remix-auth-twitter-example.na2hiro.workers.dev/

**Reply by na2hiro** [reply]
`remix-auth-twitter@2` now supports OAuth 2.0 (`Twitter2Strategy`) in addition to OAuth 1.0a (`Twitter1Strategy`)!

**Comment by joziahg** [comment]
[Google Credential Strategy](https://github.com/joziahg/remix-auth-google-credential)

Allows developers to get user Google Profile with Google Credentials via implicit oauth flow(one-tap, sign in with google)

**Comment by Lzok** [comment]
[LinkedIn Strategy](https://github.com/Lzok/remix-auth-linkedin)

**Comment by doncicuto** [comment]
[Gitea Strategy](https://github.com/doncicuto/remix-auth-gitea). [NPM Package](https://www.npmjs.com/package/@doncicuto/remix-auth-gitea)

**Comment by jrakotoharisoa** [comment]
[Okta Strategy](https://github.com/jrakotoharisoa/remix-auth-okta) 

This strategy also allow to use Okta with custom login form

**Comment by juhanakristian** [comment]
[NotionStrategy](https://github.com/juhanakristian/remix-auth-notion)

**Reply by sergiodxa** [reply]
Wow I didn’t know it was possible to do a login with Notion.

**Reply by pbteja1998** [reply]
> Wow I didn’t know it was possible to do a login with Notion.

Yes, it's possible. The API is currently in beta, but they made it possible to do a lot of things with the Notion API.

**Reply by juhanakristian** [reply]
Here's a link to the docs. https://developers.notion.com/docs/authorization#authorizing-public-integrations
NotionStrategy is based on OAuth2Strategy but Notion uses client ID and client secret as Basic Auth credentials

**Comment by neverstew** [comment]
[OpenIDConnect Strategy](https://github.com/matt-l-w/remix-auth-oidc)

**Comment by niwsa** [comment]
[BoxyHQSAMLStrategy](https://github.com/boxyhq/remix-auth-saml) for SAML. Wraps the SAML authentication in an OAuth 2.0 flow. using our open source service - https://github.com/boxyhq/jackson

**Comment by iamsamgibbs** [comment]
[Twitch Strategy](https://github.com/03gibbss/remix-auth-twitch) built off the template!

**Comment by mlcsthor** [comment]
[Keycloak Strategy](https://github.com/mlcsthor/remix-auth-keycloak)

**Comment by royanger** [comment]
[Bungie Strategy](https://github.com/royanger/remix-auth-bungie)

**Comment by JonnyBnator** [comment]
[Discord Strategy](https://github.com/JonnyBnator/remix-auth-discord)

**Comment by dev-xo** [comment]
[Time-Based One-Time Password (TOTP)](https://github.com/dev-xo/remix-auth-totp) - A TOTP Auth Strategy for Remix Auth with **Email-Code** & **Magic-Link** support.

### Highlights
- ⭐ Out-of-the-box for Remix & React Router v7  
- 📧 Magic Link login  
- ⛅ Cloudflare-ready  
- 🔐 Encrypted codes  
- 😌 Easy setup with official starters  

Give it a tryt!  
- [Live Demo](https://totp.devxo.workers.dev/)

**Comment by takagimeow** [comment]
[JWT Strategy](https://github.com/takagimeow/remix-auth-jwt)

**Comment by danphilibin** [comment]
[Twilio Verify Strategy](https://github.com/danphilibin/remix-auth-twilio)

**Comment by alexanderson1993** [comment]
[WebAuthn (Passkey) Strategy](https://www.npmjs.com/package/remix-auth-webauthn)

**Reply by sergiodxa** [reply]
OMG I have been wanting to do this for a long time, I'm going to test it on my blog deployed to Cloudflare to see if it works there too.

**Reply by alexanderson1993** [reply]
GLHF. Hopefully the instructions are clear and nothing explodes 😅

**Reply by xdivby0** [reply]
So did you try it yet, @sergiodxa ?

**Reply by alexanderson1993** [reply]
v0.5.0 is updated to support Remix Auth v4 and React Router v7.

**Comment by christopherpickering** [comment]
[SAML Strategy](https://www.npmjs.com/package/remix-auth-saml)

**Comment by sergiodxa** [comment]
- [OIDCStrategy](https://github.com/sergiodxa/web-oidc#use-with-remix-auth) from web-oidc package

**Comment by noah-guillory** [comment]
[WhoopStrategy](https://github.com/noah-guillory/remix-auth-whoop)

**Comment by cmjacques** [comment]
[Kinde Strategy](https://github.com/cmjacques/remix-auth-kinde)

**Reply by sergiodxa** [reply]
The links returns a 404, I think it’s probably private?

**Reply by cmjacques** [reply]
Sorry about that - yes, the repo was private. I intend to add some test shortly. Published because a fellow Kinde user was after a way to use Kinde with Remix. Thanks for your work on remix-auth @sergiodxa!

**Comment by alexdotpink** [comment]
[Json Strategy](https://github.com/Fumaz/remix-auth-json)

**Comment by luciengeorge** [comment]
[Salesforce Strategy](https://github.com/luciengeorge/remix-auth-salesforce)

**Comment by jmcelreavey** [comment]
Does anyone have a Jira strategy?

**Reply by sergiodxa** [reply]
There's a [Bitbucket strategy](https://github.com/doncicuto/remix-auth-bitbucket) (not sure if it's actively maintained), since both are Atlassian products, does this work for Jira too?

**Comment by bkyerv** [comment]
has anyone tried to implement auth using Telegram Login widget? I am struggling to implement just the basic one not to mention integrated with remix

**Comment by coji** [comment]
[Google Strategy for RRv7](https://github.com/coji/remix-auth-google) (forked and update)

**Reply by gregleeper** [reply]
Thanks for this fix! You might add to the readme that the user must also install OAuth2Strategy along with your package.

**Reply by chinanderm** [reply]
Thank you for forking and updating @coji!

**Reply by tatablack** [reply]
> Thanks for this fix! You might add to the readme that the user must also install OAuth2Strategy along with your package.

It's there, but in `devDependencies` rather than `dependencies`; I guess that's an oversight 🤔

**Comment by nichtsam** [comment]
[Discord Strategy for Remix Auth v4](https://github.com/nichtsam/remix-auth-discord) (forked and updated)
[Email Link Strategy for Remix Auth v4](https://github.com/nichtsam/remix-auth-email-link) (forked and updated)

**Reply by kevlened** [reply]
Thanks @nichtsam! Very helpful

**Comment by sergiodxa** [comment]
[Polar.sh Strategy](https://github.com/sergiodxa/remix-auth-polar)
